R/friendlyConstructorFunctions.R

```diff
@@ -149,3 +149,84 @@ hasSubstring <- function(string, substring) {
 
   return(v)
 }
+
+#' @export
+rowMean <- function(...) {
+  apply(data.frame(...), 1, mean, na.rm=FALSE)
+}
+
+#' @export
+rowMeanNaRm <- function(...) {
+  apply(data.frame(...), 1, mean, na.rm=TRUE)
+}
+
+#' @export
+rowSum <- function(...) {
+  apply(data.frame(...), 1, sum, na.rm=FALSE)
+}
+
+#' @export
+rowSumNaRm <- function(...) {
+  apply(data.frame(...), 1, sum, na.rm=TRUE)
+}
```

Comment:
Here is a minor optimization
```suggestion
rowMean <- function(...) {
  rowMeans(matrix(c(...), ncol = ...length()), na.rm=FALSE)
}

#' @export
rowMeanNaRm <- function(...) {
  rowMeans(matrix(c(...), ncol = ...length()), na.rm=TRUE)
}

#' @export
rowSum <- function(...) {
  rowSums(matrix(c(...), ncol = ...length()), na.rm=FALSE)
}

#' @export
rowSumNaRm <- function(...) {
  rowSums(matrix(c(...), ncol = ...length()), na.rm=TRUE)
}
```
`rowMeans` and `rowSums` are optimized versions for the very common `apply(x, 1, mean/sum)`. The R package `matrixStats` has a lot more of these (e.g., `rowQuantiles`). We probably use that package anyway so it's worthwhile to consider here.